kernel/data-pipeline/cdc/core/src/main/java/org/apache/shardingsphere/data/pipeline/cdc/core/prepare/CDCJobPrepare.java

```diff
@@ -0,0 +1,98 @@
+import org.apache.shardingsphere.data.pipeline.api.metadata.LogicTableName;
+import org.apache.shardingsphere.data.pipeline.api.metadata.loader.PipelineTableMetaDataLoader;
+import org.apache.shardingsphere.data.pipeline.api.metadata.model.PipelineColumnMetaData;
+import org.apache.shardingsphere.data.pipeline.cdc.api.CDCJobAPI;
+import org.apache.shardingsphere.data.pipeline.cdc.api.CDCJobAPIFactory;
+import org.apache.shardingsphere.data.pipeline.cdc.config.job.CDCJobConfiguration;
+import org.apache.shardingsphere.data.pipeline.cdc.config.task.CDCTaskConfiguration;
+import org.apache.shardingsphere.data.pipeline.cdc.context.job.CDCJobItemContext;
+import org.apache.shardingsphere.data.pipeline.cdc.protocol.request.CreateSubscriptionRequest.SubscriptionMode;
+import org.apache.shardingsphere.data.pipeline.core.job.PipelineJobCenter;
+import org.apache.shardingsphere.data.pipeline.core.metadata.loader.PipelineTableMetaDataUtil;
+import org.apache.shardingsphere.data.pipeline.core.prepare.InventoryTaskSplitter;
+import org.apache.shardingsphere.data.pipeline.core.task.InventoryTask;
+
+import java.util.Collection;
+import java.util.LinkedList;
+import java.util.List;
+
+/**
+ * CDC job prepare.
+ */
+@Slf4j
+public final class CDCJobPrepare {
+    
+    private final CDCJobAPI jobAPI = CDCJobAPIFactory.getInstance();
+    
+    /**
+     * Do prepare work.
+     *
+     * @param jobItemContext job item context
+     */
+    public void prepare(final CDCJobItemContext jobItemContext) {
+        if (jobItemContext.isStopping()) {
+            PipelineJobCenter.stop(jobItemContext.getJobId());
+            return;
+        }
+        CDCJobConfiguration jobConfig = jobItemContext.getJobConfig();
+        if (SubscriptionMode.FULL.name().equals(jobConfig.getSubscriptionMode())) {
+            initInventoryTasks(jobItemContext);
+        }
+        jobAPI.persistJobItemProgress(jobItemContext);
+    }
+    
+    private void initInventoryTasks(final CDCJobItemContext jobItemContext) {
+        CDCTaskConfiguration taskConfig = jobItemContext.getTaskConfig();
+        // TODO importer and channel requires a new implementation
+        InventoryDumperConfiguration inventoryDumperConfig = new InventoryDumperConfiguration(jobItemContext.getTaskConfig().getDumperConfig());
+        InventoryTaskSplitter inventoryTaskSplitter = new InventoryTaskSplitter(jobItemContext.getSourceDataSource(), inventoryDumperConfig, taskConfig.getImporterConfig());
+        List<InventoryTask> allInventoryTasks = inventoryTaskSplitter.splitInventoryData(jobItemContext);
+        jobItemContext.getInventoryTasks().addAll(allInventoryTasks);
+    }
+    
+    private Collection<InventoryDumperConfiguration> generateInventoryDumperConfigurations(final DumperConfiguration dumperConfig, final PipelineTableMetaDataLoader pipelineTableMetaDataLoader) {
+        Collection<InventoryDumperConfiguration> result = new LinkedList<>();
```

Comment:
`generateInventoryDumperConfigurations` method is not used, is it required?

Reply:
Removed now